universe/ShipDesign.cpp

```diff
@@ -677,7 +677,6 @@ void ShipDesign::BuildStatCaches() {
 
     // ensure tags are unique and copy into this->m_tags
     std::sort(tags.begin(), tags.end());
-    auto last = std::unique(tags.begin(), tags.end());
```

Comment:
Completely unused.

Reply:
the fix is to use it to avoid duplicate tags, not just remove it

Reply:
I used the usual `std::unique()` + `v.erase()` combo in this case, but I'm not sure about the later `for_each` loop. Is the following still needed or can one just copy over `tags` to `m_tags` as suggested in the original comment?
```cpp
        m_tags.push_back(std::string_view{m_tags_concatenated}.substr(next_start));
```

Reply:
```
auto next_start = m_tags_concatenated.size(); // at the end of array
m_tags_concatenated.append(str);                   // add string
// at the end of cleared m_tags add something
// that something is that concatenated array, but only starting at the end before string was added
// therefore only added string will be added to m_tags
m_tags.push_back(std::string_view{m_tags_concatenated}.substr(next_start));
```

when erasing, something could be erased from game that is not wanted

if is array unique, that also means that in original array, which is not unique, something is repeated, therefore cursor in cycle(not neccessary for_each, conservative approach may be better?) should be moved to next value in cycle...